### PR TITLE
Fetch all rows from db for Mapping list pages

### PR DESCRIPTION
**Pull request by KirtiGautam**
Added recursive calls to the APIs in generic mappings and employee mappings to fetch all the data

### CODE REVIEW

File: src/app/core/services/mappings.service.ts [hunk 1]
```diff
@@ -366,13 +366,21 @@ export class MappingsService {
...
-    return this.apiService.get(
-      `/workspaces/${workspaceId}/mappings/`, {
-        source_type: sourceType
-      }
-    );
+    const url = uri ? uri.split('api')[1] : `/workspaces/${workspaceId}/mappings/?limit=500&offset=0&source_type=${sourceType}`;
+    return this.apiService.get(url, {});
+  }
+
+  getAllMappings(sourceType) {
```
Comment (ashwin1111) [inline src/app/core/services/mappings.service.ts:375]: `sourceType: string`

File: src/app/qbo/dashboard/dashboard.component.ts [hunk 2]
```diff
@@ -112,8 +112,8 @@ export class DashboardComponent implements OnInit {
   getEmployeeMappings() {
     const that = this;
     // TODO: remove promises and do with rxjs observables
-    return that.mappingsService.getMappings('EMPLOYEE').toPromise().then((res) => {
-      if (res.results.length > 0) {
+    return that.mappingsService.getAllMappings('EMPLOYEE').toPromise().then((res) => {
```
Comment (ashwin1111) [inline src/app/qbo/dashboard/dashboard.component.ts:115]: for dashboard call why do we need to fetch all data?
something like this `getMappings(1, 0, 'EMPLOYEE')` should be good right? please tweek the service file to support this or may be add a new function. Assume if an org has 10k employees, this is unnecessary to get all data.

File: src/app/qbo/dashboard/dashboard.component.ts [hunk 3]
```diff
@@ -125,8 +125,8 @@ export class DashboardComponent implements OnInit {
   getCategoryMappings() {
     const that = this;
     // TODO: remove promises and do with rxjs observables
-    return that.mappingsService.getMappings('CATEGORY').toPromise().then((res) => {
-      if (res.results.length > 0) {
+    return that.mappingsService.getAllMappings('CATEGORY').toPromise().then((res) => {
```
Comment (ashwin1111) [inline src/app/qbo/dashboard/dashboard.component.ts:128]: same

File: src/app/core/services/mappings.service.ts [hunk 4]
```diff
@@ -366,13 +366,13 @@ export class MappingsService {
     );
   }
 
-  getMappings(sourceType, uri = null): Observable<MappingsResponse> {
+  getMappings(sourceType: string, uri:string = null, limit:number = 500): Observable<MappingsResponse> {
```
Comment (ashwin1111) [inline src/app/core/services/mappings.service.ts]: approach looks fine, ulta the last 2 order like this.
`getMappings(sourceType: string, limit: number = 500, uri: string = null): Observable<MappingsResponse> {`
I guess from dashboard you can call like this
`return that.mappingsService.getMappings('EMPLOYEE', 1).subscribe((res) => {` would look cleaner

File: src/app/core/services/mappings.service.ts [hunk 5]
```diff
@@ -366,13 +366,13 @@ export class MappingsService {
...
-    const url = uri ? uri.split('api')[1] : `/workspaces/${workspaceId}/mappings/?limit=500&offset=0&source_type=${sourceType}`;
+    const url = uri ? uri.split('api')[1] : `/workspaces/${workspaceId}/mappings/?limit=${limit}&offset=0&source_type=${sourceType}`;
     return this.apiService.get(url, {});
   }
 
-  getAllMappings(sourceType) {
+  getAllMappings(sourceType: string) {
     const that = this;
     return this.getMappings(sourceType).pipe(expand((res: any) => {
       return res.next ? that.getMappings(sourceType, res.next) : empty();
```
Comment (ashwin1111) [inline src/app/core/services/mappings.service.ts:378]: if you are changing function params, add 500 to this
`return res.next ? that.getMappings(sourceType, 500, res.next) : empty();`

File: src/app/core/services/mappings.service.ts (hunk 4 above)
Comment (KirtiGautam) [reply src/app/core/services/mappings.service.ts]: Yes, you're right

File: src/app/core/services/mappings.service.ts (hunk 5 above)
Comment (KirtiGautam) [reply src/app/core/services/mappings.service.ts:378]: Yup

File: src/app/qbo/dashboard/dashboard.component.ts [hunk 6]
```diff
@@ -125,7 +125,7 @@ export class DashboardComponent implements OnInit {
   getCategoryMappings() {
     const that = this;
     // TODO: remove promises and do with rxjs observables
-    return that.mappingsService.getMappings('CATEGORY').toPromise().then((res) => {
+    return that.mappingsService.getMappings('CATEGORY', '', 1).subscribe((res) => {
```
Comment (ashwin1111) [inline src/app/qbo/dashboard/dashboard.component.ts:128]: `''` :)

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**